template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/smm/StreamsMessagingManagerServiceConfigProvider.java

```diff
@@ -20,7 +20,7 @@
     @Override
     public List<ApiClusterTemplateConfig> getServiceConfigs(CmTemplateProcessor templateProcessor, TemplatePreparationObject source) {
         List<ApiClusterTemplateConfig> configs = Lists.newArrayList(
-                config("cm.metrics.host", source.getGeneralClusterConfigs().getClusterManagerIp()),
+                config("cm.metrics.host", source.getGeneralClusterConfigs().getExternalFQDN()),
```

Comment:
It could be null if the certification generation fails due to any reason. I think that case should also be handled here.

Reply:
If it is null, should an exception be thrown, since there isn't any default CM FQDN ? 
CM IP Address does not work for this property.

Reply:
As discussed, If externalFQDN is null, initialized the "cm.metrics.host" to CM IP as per earlier implementation.